ninja.lua

```diff
@@ -395,109 +413,63 @@ local function c_cpp_compilation_rules(cfg, toolset, pch)
 	local all_ldflags = getldflags(toolset, cfg)
 	local all_resflags = getresflags(toolset, cfg, cfg)
 
+	local _in, _out = "$in", "$out"
+
 	if toolset == p.tools.msc then
-		p.outln("CFLAGS=" .. all_cflags)
-		p.outln("rule cc")
-		p.outln("  command = " .. cc .. " $CFLAGS" .. " /nologo /showIncludes -c /Tc$in /Fo$out")
-		p.outln("  description = cc $out")
-		p.outln("  deps = msvc")
-		p.outln("")
-		p.outln("CXXFLAGS=" .. all_cxxflags)
-		p.outln("rule cxx")
-		p.outln("  command = " .. cxx .. " $CXXFLAGS" .. " /nologo /showIncludes -c /Tp$in /Fo$out")
-		p.outln("  description = cxx $out")
-		p.outln("  deps = msvc")
-		p.outln("")
-		p.outln("CFLAGS=" .. all_cflags)
-		p.outln("rule clangtidy_cc")
-		p.outln("  command = clang-tidy $in -- -x c $CFLAGS &&$")
-		p.outln("            " .. cc .. " $CFLAGS" .. " /nologo /showIncludes -c /Tc$in /Fo$out")
-		p.outln("  description = cc $out")
-		p.outln("  deps = msvc")
-		p.outln("")
-		p.outln("CXXFLAGS=" .. all_cxxflags)
-		p.outln("rule clangtidy_cxx")
-		p.outln("  command = clang-tidy $in -- -x c++ $CFLAGS &&$")
-		p.outln("            " .. cxx .. " $CXXFLAGS" .. " /nologo /showIncludes -c /Tp$in /Fo$out")
-		p.outln("  description = cxx $out")
-		p.outln("  deps = msvc")
-		p.outln("")
-		p.outln("RESFLAGS = " .. all_resflags)
-		p.outln("rule rc")
-		p.outln("  command = " .. rc .. " /nologo /fo$out $in $RESFLAGS")
-		p.outln("  description = rc $out")
-		p.outln("")
+		ninja.emit_flags("CFLAGS", all_cflags)
+		ninja.emit_rule("cc", {cc, "$CFLAGS", "/nologo", "/showIncludes", "-c", "/Tc".._in, "/Fo".._out}, "cc ".._out, {deps = "msvc"})
```

Comment:
I think I will later group flags and rules together (moving flags in `opts` or dedicated field?) and delay the print.
Then we might print rules only when used :-)